core/src/main/java/hudson/model/Computer.java

```diff
@@ -195,6 +195,8 @@
 
     protected final Object statusChangeLock = new Object();
 
+    private final Object logDirLock = new String("logDirLock);
```

Comment:
Note that thread dumps will unfortunately not show the text. Maybe a future JVM could be enhanced to do that.